mailbox/store/src/test/java/org/apache/james/mailbox/store/AbstractCombinationManagerTest.java

```diff
@@ -360,6 +364,70 @@ public void setInMailboxesFromMessageIdManagerShouldMoveMessage() throws Excepti
             .extractingResultOf("getMessageId").containsOnly(messageId);
     }
 
+    @Test
+    public void appendMessageFromMessageManagerAndSetInMailboxFromMessageIdManagerShouldUpdateApplicableFlag() throws Exception {
+        Flags messageFlag = new Flags(Flag.ANSWERED);
+        messageFlag.add(USER_FLAGS_VALUE);
+
+        MessageId messageId = messageManager1.appendMessage(new ByteArrayInputStream(MAIL_CONTENT), new Date(), session, false, messageFlag).getMessageId();
+
+        messageIdManager.setInMailboxes(messageId, ImmutableList.of(mailbox1.getMailboxId(), mailbox2.getMailboxId()), session);
+
+        Flags applicableFlags1 = messageIdManager.getApplicableFlag(mailbox1.getMailboxId(), session);
+        Flags applicableFlags2 = messageIdManager.getApplicableFlag(mailbox2.getMailboxId(), session);
```

Comment:
Applicable flags should be retrieved via MessageManager (so that we can use it in IMAP layer ;-) )